src/util/entity.cpp

```diff
@@ -193,43 +193,33 @@ namespace big::entity
 
 	bool load_ground_at_3dcoord(Vector3& location)
 	{
-		float groundZ;
-		bool done = false;
+	    constexpr float max_ground_check = 1000.f;
+	    constexpr int max_attempts = 300;
+	    float ground_z = location.z;
+	    int current_attempts = 0;
+	    bool found_ground;
 
-		for (int i = 0; i < 10; i++)
-		{
-			for (int z = 0; z < 1000; z += 25)
-			{
-				float ground_iteration = static_cast<float>(z);
-				// Only request a collision after the first try failed because the location might already be loaded on first attempt.
-				if (i >= 1 && (z % 100 == 0))
-				{
-					STREAMING::REQUEST_COLLISION_AT_COORD(location.x, location.y, ground_iteration);
-					script::get_current()->yield();
-				}
+	    do {
+	        found_ground = MISC::GET_GROUND_Z_FOR_3D_COORD(location.x, location.y, max_ground_check, &ground_z, FALSE, FALSE);
+	        STREAMING::REQUEST_ADDITIONAL_COLLISION_AT_COORD(location.x, location.y, location.z);
```

Comment:
Why did you change this out for REQUEST_ADDITIONAL_COLLISION_AT_COORD? The script only uses this in tandem with REQUEST_COLLISION_AT_COORD at the same exact coordinates.

Reply:
alright, switching back to REQUEST_COLLISION_AT_COORD